apps/molgenis-components/src/components/layout/CookieWall.vue

```diff
@@ -0,0 +1,115 @@
+<script setup>
+import { computed, ref } from "vue";
+import { useCookies } from "@vueuse/integrations/useCookies";
+import LayoutModal from "./LayoutModal.vue";
+import ButtonAction from "../forms/ButtonAction.vue";
+import ButtonAlt from "../forms/ButtonAlt.vue";
+import GTag from "./GTag.vue";
+const cookies = useCookies(["mg_allow_analytics"]);
+const props = defineProps({
+  title: {
+    type: String,
+    default: "Would you like a cookie ?",
+  },
+  acceptLabel: {
+    type: String,
+    default: "Accept",
+  },
+  rejectLabel: {
+    type: String,
+    default: "No thanks",
+  },
+  analyticsId: {
+    type: String,
+  },
+});
+
+const emit = defineEmits(["acceptCookie"]);
+
+let isDeclined = ref(false);
+
+const show = computed(() => {
+  return cookies.get("mg_allow_analytics") === undefined;
+});
+
+const cookieOptions = {
+  path: "/",
+  maxAge: 1704085200, // one year,
```

Comment:
one year seems a bit much?

Reply:
recommend default for this type of cookie